OmniKit/PumpManager/PodState.swift

```diff
@@ -104,6 +104,8 @@ public struct PodState: RawRepresentable, Equatable, CustomDebugStringConvertibl
     
     public var insulinType: InsulinType
     
+    public var skipNextCommsOptimization: Bool
```

Comment:
Having control-flow variables at this wide scope strikes me as a bad pattern. It creates a situation where the code branches based on whether it is set or not, but the name offers no clue as to *why* the branch is going one way or the other.

State variables should convey something about the state of the program, and describe the state, rather than describing how code should flow. The code can branch in reaction to the state.

Is there some way you can change this to describe the actual state you're wanting to modify behavior for?  I.e. if you want to do something special when the last request errored, it'd be much better to store a variable called `lastRequestErrored`, than a flag called `doSomethingSpecial`.  And then the code can read

```
if lastRequestDidError {
    doSomethingSpecial()
}
```

Reply:
The original prototype this variable was actually named `lastCommsFailed`. But this variable's use has evolved and it is now set for the following three distinct cases and cleared only upon Loop receiving an expected response:

+ Whenever Loop doesn't receive the expected response for any message it sends
+ Whenever the pod's bolus or temp basal delivery status doesn't match Loop's
+ On Loop app start up

For all of these cases, any pod comms optimizations (i.e., skipping the get status command before a bolus and later skipping the cancel TB command when going from the scheduled basal to a TB) should not be performed for the best/natural behavior in the face of unreliable comms. Thus I thought that using `skipNextCommsOptimization` was a good way to succinctly describe this variable's use (and upon reflection it actually describe the state of the program). If you still don't like the variable name which I decided on after some thought (and with no consultation which I had requested), please suggest a suitable alternate variable name which would encapsulate its various uses/state as I'd like to get this PR to land ASAP. Thanks!

Would a comment on the variable declaration documenting the various places it is used be helpful?